ring/basic_lifecycler.go

```diff
@@ -63,12 +63,17 @@ type BasicLifecyclerConfig struct {
 	RingTokenGenerator TokenGenerator
 }
 
-// BasicLifecycler is a basic ring lifecycler which allows to hook custom
-// logic at different stages of the lifecycle. This lifecycler should be
-// used to build higher level lifecyclers.
-//
-// This lifecycler never change the instance state. It's the delegate
-// responsibility to ChangeState().
+/*
+BasicLifecycler is a Service that is responsible for publishing changes to a ring for a single instance.
```

Comment:
I've tried to use `[services.Service]` or `[github.com/grafana/dskit/services.Service]` links here, but links were always broken in godoc :( Too bad.

Reply:
Same - I tried a few different things, but couldn't get it to work :/

Reply:
It looks like imported package links don't work with godoc [due to a bug](https://github.com/golang/go/issues/56683) 🤦